### DISCUSSION TITLE
PCButton Example Sends Pan Control MIDI with No Pot Connected

### DISCUSSION THREAD

**Discussion by MickeyGentry** [tttapa/Control-Surface#915] [General]
I'm using an Arduino Micro with external power and USB plugged into a Mac. I installed the PCButton.ino example which works, the program change MIDI command is sent from a button press. But there is constant stream of controller pan position commands along with the button press command. This is seen in MidiView app:
![Screenshot 2023-04-12 at 3 58 09 PM](https://user-images.githubusercontent.com/130603709/231570635-cb22a45f-e695-49cf-814f-dbd2d9ec8783.png)

This also happens with no button or anything but power and USB connected.
What am I missing? Is this a sort of 'bounce' problem?

Thank you!

**Comment by tttapa** [comment]
I can't seem to be able to reproduce this on my end, and I'm not sure what could cause this. Do you observe the same problem using a `USBDebugMIDI_Interface` instead of `USBMIDI_Interface`?

**Reply by MickeyGentry** [reply]
I have not but I will try that, thanks

**Reply by MickeyGentry** [reply]
Well I tried the sketch with the debug interface, now there's no output at all, button or pan. Went back and forth between debug MIDI and usual MIDI, one works with unwanted output and one does nothing. 

Obviously I'm new at this but it's odd, no?

**Reply by tttapa** [reply]
The debug interface prints to the Arduino Serial Monitor, it won't send actual MIDI messages.

**Comment by MickeyGentry** [comment]
Ah, thanks for the schooling.Sent from my iPadOn Apr 12, 2023, at 5:27 PM, Pieter P ***@***.***> wrote:﻿
The debug interface prints to the Arduino Serial Monitor, it won't send actual MIDI messages.

**Comment by MickeyGentry** [comment]
So I may have ruined the board. I was using running the PCButton example with a temporary button WITHOUT a resistor in series (all my bad). It did work for a while but now the board's blue light is on with external power but no flashing LEDs on power up and no computer connection. 

Think the board is dead?

Thanks for your help!

**Reply by MickeyGentry** [reply]
Hm, I see the internal pull-up resistor is enabled in code for PCbutton, so I don't know why this board seems dead.

**Reply by tttapa** [reply]
Like you said, the internal pull-up resistor is enabled by the software, so no external resistor is required. (The resistor you need is also not a series resistor, those are necessary for LEDs, but not switches.)

Unless you shorted any pins with it or enabled `OUTPUT` mode in your code, the switch is unlikely to be the root cause.  
Do the LEDs blink if you press the reset button?

**Comment by MickeyGentry** [comment]
Sadly nothing happens. The only light on power up is the blue LED, no lights on pressing reset.I think I left the switch connected, loaded a very basic sketch and pressed it, which I assume would short out.Sent from my iPadOn Apr 14, 2023, at 4:06 PM, Pieter P ***@***.***> wrote:﻿
Like you said, the internal pull-up resistor is enabled by the software, so no external resistor is required. (The resistor you need is also not a series resistor, those are necessary for LEDs, but not switches.)
Unless you shorted any pins with it or enabled OUTPUT mode in your code, the switch is unlikely to be the root cause.
Do the LEDs blink if you press the reset button?

**Comment by MickeyGentry** [comment]
Hey, just to close this out, got a new board and hooked it up _correctly_ and everything works fine. Thanks!

**Reply by tttapa** [reply]
Glad to hear, thanks for the update!
